From 93c8bd7265b3eca63d3b649ac95c60249fa39512 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Thu, 27 Apr 2023 13:18:17 +0800
Subject: [PATCH 59/68] Revert "drm/bridge: megachips: Fix error handling in
 i2c_register_driver()"

This reverts commit 5b9bcb33cf3ffa60c520c02b33a364cb64d7e7df.
---
 drivers/gpu/drm/bridge/megachips-stdpxxxx-ge-b850v3-fw.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/bridge/megachips-stdpxxxx-ge-b850v3-fw.c b/drivers/gpu/drm/bridge/megachips-stdpxxxx-ge-b850v3-fw.c
index e41afcc5326b1..72248a565579e 100644
--- a/drivers/gpu/drm/bridge/megachips-stdpxxxx-ge-b850v3-fw.c
+++ b/drivers/gpu/drm/bridge/megachips-stdpxxxx-ge-b850v3-fw.c
@@ -444,11 +444,7 @@ static int __init stdpxxxx_ge_b850v3_init(void)
 	if (ret)
 		return ret;
 
-	ret = i2c_add_driver(&stdp2690_ge_b850v3_fw_driver);
-	if (ret)
-		i2c_del_driver(&stdp4028_ge_b850v3_fw_driver);
-
-	return ret;
+	return i2c_add_driver(&stdp2690_ge_b850v3_fw_driver);
 }
 module_init(stdpxxxx_ge_b850v3_init);
 
-- 
2.25.1

